python/python/lance/dataset.py

```diff
@@ -2997,6 +3041,7 @@ class Transaction:
     operation: LanceOperation.BaseOperation
     uuid: str = dataclasses.field(default_factory=lambda: str(uuid.uuid4()))
     blobs_op: Optional[LanceOperation.BaseOperation] = None
+    transaction_properties: Optional[Dict[str, str]] = None
```

Comment:
I think this can just default to an empty dict.
```suggestion
    transaction_properties: Dict[str, str] = dataclasses.field(default_factory=dict)
```

Reply:
Thanks for the review! Good call — I’ll update this to default to an empty dict.